x-pack/plugins/canvas/public/components/app/index.tsx

```diff
@@ -28,12 +31,21 @@ class ShortcutManagerContextWrapper extends React.Component<React.PropsWithChild
   }
 }
 
-export const App: FC<{ history: ScopedHistory }> = ({ history }) => {
-  const { updatePath } = useNavLinkService();
-
+export const App: FC<{ history: ScopedHistory; appUpdater: BehaviorSubject<AppUpdater> }> = ({
+  history,
+  appUpdater,
+}) => {
   useEffect(() => {
     return history.listen(({ pathname, search }) => {
-      updatePath(pathname + search);
+      const path = pathname + search;
+      appUpdater.next(() => ({
+        defaultPath: `${path}`,
```

Comment:
This cange be just `defaultPath: path`

Reply:
Totally - I copy pasted this from a different file but didn't catch that. Thanks! Fixed in 78ee22bf218055686f3659cf75fd8bfb7b494504